src/dirwatcher.js

```diff
@@ -0,0 +1,90 @@
+import EventEmmiter from 'events';
+import fs from 'fs';
+import { promisify } from 'util';
+import crypto from 'crypto';
+
+const readFile = promisify(fs.readFile);
+const readdir = promisify(fs.readdir);
+
+export class DirWatcher extends EventEmmiter {
+  constructor() {
+    super();
+    this.isFirstCheck = true;
+    this.prevFiles = [];
+    this.prevFilesList = [];
+  }
+
+  watch(path, delay) {
+    this.isFirstCheck ? 
+      this.firstReading(path)
+        .then((files) => {
+          this.prevFiles = files;
+          this.isFirstCheck = false;
+        }) : 
+      this.compareFiles(path)
+        .then((files) => {
+          this.prevFiles = files;
+        });
+    setTimeout(() => {
+      console.log('check completed, watch')
```

Comment:
Could you please remove unnecessary console.logs

Reply:
Done